smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HttpProtocolClientGenerator.kt

```diff
@@ -41,6 +41,11 @@ open class HttpProtocolClientGenerator(
         writer.addImport(SwiftDependency.CLIENT_RUNTIME.target)
         writer.addImport(SwiftDependency.SWIFT_LOG.target)
         writer.addFoundationImport()
+        if (ctx.protocol.name.lowercase().contains("json")) {
+            writer.addImport(SwiftDependency.JSON_RUNTIME.target)
+        } else {
+            writer.addImport(SwiftDependency.XML_RUNTIME.target)
+        }
```

Comment:
1.  This code seems to be pretty repetitive. Can we add something to SwiftWriter like:
```
fun addSerdeDependency(protocol: ShapeId) {
    if (json) {

    } else if xml {
        ...
    }
```
I believe this can be used in cases we have either:
a: where we have common code for both json/xml like:
```
      if (ctx.protocol.name.lowercase().contains("json")) {
                    writer.addImport(SwiftDependency.JSON_RUNTIME.target)
                } else {
                    writer.addImport(SwiftDependency.XML_RUNTIME.target)
                }
```
b. where we have specific code for just xml like:
```
writer.addImport(SwiftDependency.XML_RUNTIME.target)
```
That way, we can have a common function to inject the correct import and not have to make changes across the codebase.



2. Note the `else if` above vs `else`.  I think it would be better to be explicit about which serde dependency we need.